experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -157,278 +156,126 @@ function composeMarkLists<TNodeChange>(
-					return 0;
-				}
-				case "MoveOut": {
-					if (baseMark.detachedBy === newIntention) {
-						getOrAddEffect(
-							moveEffects,
-							CrossFieldTarget.Source,
-							baseMark.revision,
-							baseMark.id,
-							true,
-						).shouldRemove = true;
-						getOrAddEffect(
-							moveEffects,
-							CrossFieldTarget.Destination,
-							newMark.revision ?? newRev,
-							newMark.id,
-							true,
-						).shouldRemove = true;
-						return 0;
-					} else {
-						getOrAddEffect(
-							moveEffects,
-							CrossFieldTarget.Source,
-							baseMark.revision,
-							baseMark.id,
-							true,
-						).mark = composeMark(newMark, newRev, composeChild);
-						getOrAddEffect(
-							moveEffects,
-							CrossFieldTarget.Destination,
-							newMark.revision ?? newRev,
-							newMark.id,
-							true,
-						);
-						return 0;
-					}
-				}
-				case "ReturnFrom": {
-					if (
-						baseMark.detachedBy === newIntention ||
-						newMark.detachedBy === baseIntention
-					) {
-						getOrAddEffect(
-							moveEffects,
-							CrossFieldTarget.Source,
-							baseMark.revision,
-							baseMark.id,
-							true,
-						).shouldRemove = true;
-						getOrAddEffect(
-							moveEffects,
-							CrossFieldTarget.Destination,
-							newMark.revision ?? newRev,
-							newMark.id,
-							true,
-						).shouldRemove = true;
-						return 0;
-					} else {
-						if (newMark.changes !== undefined) {
-							getOrAddEffect(
-								moveEffects,
-								CrossFieldTarget.Source,
-								baseMark.revision,
-								baseMark.id,
-								true,
-							).modifyAfter = newMark.changes;
-						}
-						getOrAddEffect(
-							moveEffects,
-							CrossFieldTarget.Source,
-							baseMark.revision,
-							baseMark.id,
-							true,
-						).mark = composeMark(newMark, newRev, composeChild);
-						getOrAddEffect(
-							moveEffects,
-							CrossFieldTarget.Destination,
-							newMark.revision ?? newRev,
-							newMark.id,
-							true,
-						);
-						return 0;
-					}
-				}
-				default:
-					fail("Not implemented newType");
-			}
+
+		if (moveOutId !== undefined) {
+			assert(isMoveMark(newMark), "Only move marks have move IDs");
+
+			// The nodes attached by `baseMark` have been moved by `newMark`.
+			// We can represent net effect of the two marks by moving `baseMark` to the destination of `newMark`.
+			getOrAddEffect(
+				moveEffects,
+				CrossFieldTarget.Destination,
+				newMark.revision ?? newRev,
+				newMark.id,
+				true,
+			).mark = withNodeChange(baseMark, nodeChange);
+			return 0;
+		} else if (isNewAttach(baseMark) && baseMark.revision === undefined) {
+			// This case is to support squashing an attach and detach in the same transaction.
+			assert(nodeChange === undefined, "TODO: Support transient inserts");
+			return 0;
 		}
-		default:
-			fail("Composing this baseType and this newType is not implemented");
+		// TODO: Create modify mark for transient node.
```

Comment:
It looks like there are already tests which run into this.

Reply:
Since this PR is not trying to add support for changes to removed content, we should indeed just return 0 and leave a TODO.
Same for the `else if (isNewAttach(baseMark) && baseMark.revision === undefined)` branch above, which perhaps can just ben removed.